src/main.rs

```diff
@@ -466,6 +513,7 @@ impl RustyShare {
         let (parts, body) = req.into_parts();
         match (&parts.method, parts.uri.path()) {
             (&Method::GET, "/") => self.index(),
+            (&Method::GET, "/register") | (&Method::POST, "/register") => self.register_or_login(parts, body),
```

Comment:
It's probably fine to keep this as `register` and return 404 if a user already exists.

Reply:
OK